lib/variable.js

```diff
@@ -79,7 +79,7 @@ module.exports = function environmentVariable (name, value) {
         );
       }
 
-      return Boolean(val);
+      return Boolean('false' === val ? 0 : val);
```

Comment:
Also, could we make the check not be case sensitive, perhaps like the below code?

```js
if (val && typeof val === 'string') {
  return Boolean('false' === val.toLowerCase() ? 0 : val);
} else {
  return Boolean(val)
}
```